Wrappers/Python/cil/utilities/quality_measures.py

```diff
@@ -18,36 +18,90 @@
 # CIL Developers, listed at: https://github.com/TomographicImaging/CIL/blob/master/NOTICE.txt
 
 from cil.optimisation.functions import L2NormSquared, L1Norm
+from cil.framework import DataContainer
 import numpy as np
 
 
-def mse(dc1, dc2):    
+
+def mse(dc1, dc2, mask=None):    
     
     ''' Returns the Mean Squared error of two DataContainers
-    '''
     
-    diff = dc1 - dc2    
-    return L2NormSquared().__call__(diff)/dc1.size
+    Parameters
+    ----------
+    dc1: `DataContainer`
+        One image to be compared
+    dc2: `DataContainer`
+        Second image to be compared 
+    mask: array or `DataContainer` of Boolean values or 0's and 1's with the same dimensions as the `dc1` and `dc2`
+        Region of interest for the calculation. 
+    '''  
+
+    if mask is None:
+        diff = dc1 - dc2    
+        return L2NormSquared().__call__(diff)/dc1.size
+    else:
+        if isinstance(mask, DataContainer):
+            mask = mask.as_array()
+        return np.mean(((dc1.as_array() - dc2.as_array())**2), where=mask.astype('bool'))
```

Comment:
numpy subtract and square probably take `where` too. Currently you are doing those computations across all pixels.

Or, as long as the operation is pixel wise, maybe it's best to use `numpy.where` on both the datacontainers first and then do the calculation as usual on the resulting arrays.

This applies to the other functions too.

Reply:
I think I have done this - thanks